### DISCUSSION TITLE
Solved: JPG to PNG. Set Transparency from Photoshop Path

### DISCUSSION THREAD

**Discussion by HeinziTuberkel** [ImageMagick/ImageMagick#2797] [Help]
Hi,
I have several (>1000) .jpg files that contain a clipping path made with Adobe Photoshop.
Now I'm supposed to 
 a) convert them to .png 
 b) make the background outside the path transparent
 c) clip the surrounding area
I had thought, with ImageMagick this must be a simple everyday task, but I can't get it to work. 
After some hours of reading and trying I gave up when the following command left me with an emty picture:
magick \temp\g2.jpg -alpha transparent -clip -alpha opaque \temp\g.png

**Comment by snibgo** [comment]
What version of IM, on what platform?

Your command works as expected for me, on a JPG with clipping path, and IM v7.0.8-64 on Windows 8.1.

To help diagnose your problem, please post a link to a sample input JPG. If you can't do that, then paste the text output from "magick \temp\g2.jpg -verbose info:"

**Reply by HeinziTuberkel** [reply]
Thanks for your quick reply.
While waiting i kept trying. I gained some more understanding of ImageMagick and found a solution. 
It took me a while to understand, that I can queue several conversion commands as parameters in one call.

So now my new command works and it looks like this:
magick  \temp\g2.jpg -alpha set -channel alpha -clip -negate +clip -channel alpha -negate -trim  \temp\g.png

BTW: I'm using ImageMagick-7.0.9-Q16 on Windows 10 Pro.

**Reply by fmw42** [reply]
That bug was fixed, I believe, in later versions. Try upgrading to the current version.
